### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by topeogunleye**
### Added Changes

- Created a User Interface for the application
- Added and Implemented: All methods are visible in the diagram.
- Created a schema.sql file with tables that will be analogical to the structure of the classes in your app.
- Added Tests for all methods.

NOTE TO REVIEWER, We have three people in this group, however, the third person's GitHub picture does not show up because he changed his GitHub email address.

[Video Demo](https://www.youtube.com/watch?v=YmnojSJ2uHA&feature=youtu.be )

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+CREATE TABLE musicalbum (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(100),
+    publish_date DATE,
+    archived BOOLEAN,
+    on_spotify BOOLEAN DEFAULT 'f',
+    genre_id INT,
+    CONSTRAINT genre_fk FOREIGN KEY (genre_id) REFERENCES genre(id)
+);
```
Comment (julie-ify) [inline database/schema.sql:47]: - [ ] According to the ERD, musicalbum has only the `id` and `on_spotify` columns. Kindly remove every other column here, please.

```suggestion
CREATE TABLE musicalbum (
    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
    on_spotify BOOLEAN,
    FOREIGN KEY(item_id) REFERENCES items(id)
);
```

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    lable_id INT NOT NULL,
+    source_id INT NOT NULL,
+    author_id INT NOT NULL,
+    publish_date DATE NOT NULL,
+    archived BOOLEAN NOT NULL,
+    FOREIGN KEY genre_id REFERENCES genre (id),
+    FOREIGN KEY lable_id REFERENCES lable (id),
+    FOREIGN KEY author_id REFERENCES author (id),
+    FOREIGN KEY source_id REFERENCES source (id)
+);
```
Comment (julie-ify) [inline database/schema.sql:77]: - [ ] According to the ERD, the games table has only the `id`, `multiplayer` and `last_played_at` columns. Kindly remove every other column and the foreign keys here, please.

```suggestion
CREATE TABLE games (
    id INT PRIMARY KEY,
    multiplayer VARCHAR(100),
    last_played_at DATE,
    FOREIGN KEY(item_id) REFERENCES items(id)
);
```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Paskab012, @vikipretium and @topeogunleye,

Good job so far!

## Highlights!
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Good code organization ✔️
- Descriptive README ✔️ 
- Good video presentation ✔️ 
- Tests passed ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  def add_game
+    multiplayer, last_played_at = game_input
+    publish_date = publish_date_input
+
+    @games << Game.new(multiplayer, last_played_at, publish_date)
+    save_games
+    puts 'Game created successfully'
+    puts ''
+  end
```
[the remark this answers is not in the document]
Comment (vikipretium) [reply app.rb:101]: we fixed it @julie-ify 👍🏼

File: database/schema.sql [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    title VARCHAR(100) NOT NULL,
+    color VARCHAR(100) NOT NULL,
+);
+
+CREATE TABLE labels (
+    id  INT GENERATED ALWAYS AS IDENTITY,
+    title VARCHAR(100),
+    color VARCHAR(100),
+    PRIMARY KEY(id)
+);
```
[the remark this answers is not in the document]
Comment (vikipretium) [reply database/schema.sql]: @julie-ify I removed all the duplicates and thanks for helping us to fix our project 👍🏼

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    @games = load_games
+    @authors = load_authors
+  end
+
+  def display_menu
+    puts 'Select an option: '
+    puts
+    puts ['1 - List Books', '2 - List Music Albums', '3 - List Games', '4 - List Genres', '5 - List Labels',
+          '6 - List Authors', '7 - Add Book', '8 - Add Music Album', '9 - Add Game', '10 - Exit App']
+  end
```
Comment (Emmyn5600) [inline app.rb:23]: - [ ] When you click on the first 6 options, nothing is displayed when there are no data inserted already, so let us display a clear to the users when a user chose one of those menus before entering the data.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @topeogunleye @vikipretium  and @Paskab012 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 5 above)
Comment (Emmyn5600) [reply app.rb:23]: Hello, @topeogunleye @vikipretium @Paskab012 I can see the message that is being displayed when a user chose to see or check the list of items when there is no data added.

image

Comment (vikipretium) [reply app.rb:23]: Thanks, @Emmyn5600 👍🏼 We are grateful

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Thanks for working hard to make the changes requested by the previous reviewer, it's time to merge and drink coffee 🎉

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**